src/mon/AuthMonitor.cc

```diff
@@ -1155,6 +1155,12 @@ bool AuthMonitor::prepare_command(MonOpRequestRef op)
     }
     err = 0;
 
+    //if capability strings are malformed, return with error msg.
+    if (!valid_caps(caps_vec, &ss)) {
```

Comment:
Please move this check to right after the vector is populated. That would be right after the block ending in line 1044.

Reply:
@jecluis Done.